### PR TITLE
refactor: tests with microservice

### PR DESCRIPTION
**Pull request by KubaBaniak**
## Refactor tests for this PR: https://github.com/KubaBaniak/RecipeAppNestjs/pull/70

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,6 +8,11 @@ defaults: &defaults
     - image: cimg/node:18.16.0
     - image: cimg/postgres:14.1
     - image: redis:7.0.12-alpine
+    - image: rabbitmq:3.13.0-rc.1-alpine
```
Comment (rad20c) [inline .circleci/config.yml:11]: I wouldn't use release candidate version of images - I'd recommend switching to `3.12.8-alpine`

File: src/user/test/user.service.spec.ts [hunk 2]
```diff
@@ -28,15 +28,16 @@ describe('UserService', () => {
   describe('CreateUser', () => {
     it('should create default User', async () => {
       //given
-      const request = createUser();
+      const { email } = createUser();
+      const password = faker.internet.password({ length: 64 });
 
       //when
-      const createdUser = await userService.createUser(request);
+      const createdUser = await userService.createUser({ email, password });
```
Comment (rad20c) [inline src/user/test/user.service.spec.ts:35]: hmm, why did we remove the `password` from `createUser` if we need it anyway?

File: src/auth/test/auth.service.spec.ts [hunk 3]
```diff
@@ -201,106 +131,110 @@ describe('AuthService', () => {
...
+      const generatedRecoveryKeys = await authService.generate2faRecoveryKeys(
+        userId,
+      );
+
+      expect(generatedRecoveryKeys).toEqual(recoveryKeys);
     });
   });
 
   describe('Enable 2fa', () => {
     it('should enable 2fa', async () => {
```
Comment (rad20c) [inline src/auth/test/auth.service.spec.ts:202]: You could do `ctrl + f` in IDE and find all instances of `"2fa"` as strings / descriptions etc. and change them to `"2FA"` ("2FA" in strings, "2fa" in object / class / variable names etc.)

File: test/webhook.e2e-spec.ts [hunk 4]
```diff
@@ -8,23 +8,23 @@ import { User } from '@prisma/client';
...
 describe('WebhookController (e2e)', () => {
   let app: INestApplication;
-  let authService: AuthService;
   let prismaService: PrismaService;
   let user: User;
   let accessToken: string;
+  let jwtService: JwtService;
 
-  beforeAll(async () => {
+  beforeEach(async () => {
```
Comment (rad20c) [inline test/webhook.e2e-spec.ts:27]: Why did you change it to `beforeEach`? I think testing module should be created once. Now you are creating it for each test.

**Review by rad20c — APPROVED** [review]

File: src/user/test/user.service.spec.ts (hunk 2 above)
Comment (KubaBaniak) [reply src/user/test/user.service.spec.ts:35]: I used this function a lot while creating data for auth tests. But now since the `db` here (of base service) does not contain user password, and most of the endpoints did not required password after that change I removed it.

File: test/webhook.e2e-spec.ts (hunk 4 above)
Comment (KubaBaniak) [reply test/webhook.e2e-spec.ts]: I've noticed that I did not have a call to database that deletes all webhooks and its events before. So to fix that, what I did is change it so it runs multiple testing modules. 

But I changed it, so test module is created once, and records created in test are being deleted after.
